### DISCUSSION TITLE
PHP Fatal error:  Trait 'setasign\Fpdi\FpdiTrait' not found in ../../../vendor/mpdf/mpdf/src/FpdiTrait.php on line 25

### DISCUSSION THREAD

**Discussion by gsdandial** [mpdf/mpdf#1825] [Q&A]
i am using mpdf package for creating pdf file with PHP. on my Local server i have downloaded this package. but when i have uploaded on server , its giving me this error : 
 PHP Fatal error:  Trait 'setasign\Fpdi\FpdiTrait' not found in ../vendor/mpdf/mpdf/src/FpdiTrait.php on line 25

plz provide me help , i have searched on google , but got no information regarding this.
**first 30 lines of Fditrait.php** 
```
<?php

namespace Mpdf;

use setasign\Fpdi\PdfParser\CrossReference\CrossReferenceException;
use setasign\Fpdi\PdfParser\Filter\AsciiHex;
use setasign\Fpdi\PdfParser\Type\PdfArray;
use setasign\Fpdi\PdfParser\Type\PdfDictionary;
use setasign\Fpdi\PdfParser\Type\PdfHexString;
use setasign\Fpdi\PdfParser\Type\PdfIndirectObject;
use setasign\Fpdi\PdfParser\Type\PdfIndirectObjectReference;
use setasign\Fpdi\PdfParser\Type\PdfName;
use setasign\Fpdi\PdfParser\Type\PdfNull;
use setasign\Fpdi\PdfParser\Type\PdfNumeric;
use setasign\Fpdi\PdfParser\Type\PdfStream;
use setasign\Fpdi\PdfParser\Type\PdfString;
use setasign\Fpdi\PdfParser\Type\PdfType;
use setasign\Fpdi\PdfParser\Type\PdfTypeException;
use setasign\Fpdi\PdfReader\DataStructure\Rectangle;
use setasign\Fpdi\PdfReader\PageBoundaries;

/**
 * @mixin Mpdf
 */
trait FpdiTrait
{
	use \setasign\Fpdi\FpdiTrait {
		writePdfType as fpdiWritePdfType;
		useImportedPage as fpdiUseImportedPage;
		importPage as fpdiImportPage;
	}
```

File which calling mpdf package to generate PDF file
```


[
$defaultFontConfig = (new Mpdf\Config\FontVariables())->getDefaults();
$fontData = $defaultFontConfig['fontdata'];

$mpdf = new \Mpdf\Mpdf(['fontdata' => $fontData + [ // lowercase letters only in font key
    'anmollipi' => [
        'R' => 'AnmolLipi.ttf',
        //   'I' => 'FrutigerObl-Normal.ttf',
    ]
], 'format' => $size, 'orientation' => $orientation]);
$mpdf->tabSpaces = 6;
$mpdf->setHTMLFooter($footer);
$mpdf->WriteHTML($paper_data);
if ($mpdf->Output($file_name)){](url)

```

**Comment by finwe** [comment]
Have you uploaded the entire vendor directory to the server? Can't you run composer install on the server?

**Reply by gsdandial** [reply]
Yes i have uploaded mpdf folder from vendor directory and composer folder along with all composer related files

**Reply by finwe** [reply]
is there opcache that should be cleared?

**Reply by hdutta1984** [reply]
@gsdandial Did you able to fix the issue?

**Comment by hdutta1984** [comment]
I am also facing the issue. Kindly help how to solve this issue.

Fatal error: Trait "setasign\Fpdi\FpdiTrait" not found in \vendor\mpdf\mpdf\src\FpdiTrait.php on line 25

**Comment by b4rt3kk** [comment]
I've had the same issue. Well, that occurs when you work with remote server, while installing packages locally and synchronization fails.

Normally in that way, you just reupload newly installed package, and autoloader files. 

But mPDF have dependant packages. This certain error occurs because you don't manually reuploaded `vendor/setasign` package.

**Comment by findshorty** [comment]
For anyone still struggling with this, you need to update to php 8.4+
